packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2098,6 +2123,26 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   @protected
   bool get alwaysNeedsCompositing => false;
 
+  late bool _wasRepaintBoundary;
+
+  /// Update the composited layer owned by this render object.
+  ///
+  /// This method is called by the framework for render object repaint
+  /// boundaries to update the properties on their composited layer, potentially
+  /// without repainting their children.
+  ///
+  /// If [oldLayer] is `null`, this method should return a new [OffsetLayer]
+  /// (or subtype thereof). If [oldLayer] is not `null`, then this method should
+  /// reuse the layer instance that is provided - it is an error to create a new
+  /// layer in this instance.
+  ///
+  /// The [OffsetLayer.offset] property will be managed by the framework and
```

Comment:
need not -> must not? Or what happens if I do update it? Will it just get ignored?

Reply:
done

Reply:
well, I said must not and added an assert....